[Automation] javac compile fails for org.apache.maven.plugins:maven-source-plugin:3.3.0

## javac compile on ubuntu-latest / JDK 25 / mode current-defaults

Failure kind: javac compile
OS: ubuntu-latest
JDK: 25
Native-image mode: current-defaults
Reproducer: `env GVM_TCK_LV=3.3.0 GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean compileTestJava -Pcoordinates=org.apache.maven.plugins:maven-source-plugin:3.2.0`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/26389440080
Last 300 lines of the log (excerpt truncated to fit the GitHub issue size limit):
```console
e-plugin:3.2.0:3.3.0 stage javac compile
========================================================================================
Starting stage 'javac compile' for 3.3.0 at 2026-05-25T07:52:57Z
Command: env GVM_TCK_LV=3.3.0 GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean compileTestJava -Pcoordinates=org.apache.maven.plugins:maven-source-plugin:3.2.0
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 823ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :compileTestJava FAILED
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :tck-test-support:compileJava UP-TO-DATE

/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:9: error: package org.apache.maven.plugin.logging does not exist
> Task :compileTestJava FAILED
import org.apache.maven.plugin.logging.Log;
                                      ^
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:10: error: cannot find symbol
import org.apache.maven.plugins.source.HelpMojo;
                                      ^
  symbol:   class HelpMojo
  location: package org.apache.maven.plugins.source
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:31: error: cannot find symbol
    private static final class CapturingLog implements Log {
                                                       ^
  symbol:   class Log
  location: class HelpMojoTest
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:18: error: cannot find symbol
        HelpMojo mojo = new HelpMojo();
        ^
  symbol:   class HelpMojo
  location: class HelpMojoTest
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:18: error: cannot find symbol
        HelpMojo mojo = new HelpMojo();
                            ^
  symbol:   class HelpMojo
  location: class HelpMojoTest
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:45: error: method does not override or implement a method from a supertype
        @Override
        ^
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:50: error: method does not override or implement a method from a supertype
        @Override
        ^
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:55: error: method does not override or implement a method from a supertype
        @Override
        ^
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:61: error: method does not override or implement a method from a supertype
        @Override
        ^
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:66: error: method does not override or implement a method from a supertype
        @Override
        ^
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:71: error: method does not override or implement a method from a supertype
        @Override
        ^
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:76: error: method does not override or implement a method from a supertype
        @Override
        ^
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:82: error: method does not override or implement a method from a supertype
        @Override
        ^
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:87: error: method does not override or implement a method from a supertype
        @Override
        ^
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:92: error: method does not override or implement a method from a supertype
        @Override
        ^
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:97: error: method does not override or implement a method from a supertype
        @Override
        ^
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:103: error: method does not override or implement a method from a supertype
        @Override
        ^
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:108: error: method does not override or implement a method from a supertype
        @Override
        ^
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:113: error: method does not override or implement a method from a supertype
        @Override
        ^
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:118: error: method does not override or implement a method from a supertype
        @Override
        ^
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:124: error: method does not override or implement a method from a supertype
        @Override
        ^
21 errors

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed; see the compiler output below.
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:10: error: cannot find symbol
  import org.apache.maven.plugins.source.HelpMojo;
                                        ^
    symbol:   class HelpMojo
    location: package org.apache.maven.plugins.source
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:31: error: cannot find symbol
      private static final class CapturingLog implements Log {
                                                         ^
    symbol:   class Log
    location: class HelpMojoTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:18: error: cannot find symbol
          HelpMojo mojo = new HelpMojo();
          ^
    symbol:   class HelpMojo
    location: class HelpMojoTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:18: error: cannot find symbol

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/build/reports/problems/problems-report.html
          HelpMojo mojo = new HelpMojo();
                              ^
    symbol:   class HelpMojo
    location: class HelpMojoTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:45: error: method does not override or implement a method from a supertype
          @Override
          ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:50: error: method does not override or implement a method from a supertype
          @Override
          ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:55: error: method does not override or implement a method from a supertype
          @Override
          ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:61: error: method does not override or implement a method from a supertype
          @Override
          ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:66: error: method does not override or implement a method from a supertype
          @Override
          ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:71: error: method does not override or implement a method from a supertype
          @Override
          ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:76: error: method does not override or implement a method from a supertype

          @Override
          ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:82: error: method does not override or implement a method from a supertype
          @Override
          ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:87: error: method does not override or implement a method from a supertype
          @Override
          ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:92: error: method does not override or implement a method from a supertype
          @Override
          ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:97: error: method does not override or implement a method from a supertype
          @Override
          ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:103: error: method does not override or implement a method from a supertype
          @Override
          ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:108: error: method does not override or implement a method from a supertype
          @Override
          ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:113: error: method does not override or implement a method from a supertype
          @Override
          ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:118: error: method does not override or implement a method from a supertype
          @Override
          ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:124: error: method does not override or implement a method from a supertype
          @Override
          ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:9: error: package org.apache.maven.plugin.logging does not exist
  import org.apache.maven.plugin.logging.Log;
                                        ^
  21 errors

* Try:
> Check your code and dependencies to fix the compilation error(s)
> Run with --scan to generate a Build Scan (Powered by Develocity).
Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
10 actionable tasks: 1 executed, 9 up-to-date

BUILD FAILED in 1s

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 2s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'javac compile' for 3.3.0 at 2026-05-25T07:53:00Z with exit code 1 after 3s
❌ ERROR: 'javac compile' for 3.3.0 failed with exit code 1.
FAILED[javac compile][3.3.0][env GVM_TCK_LV=3.3.0 GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean compileTestJava -Pcoordinates=org.apache.maven.plugins:maven-source-plugin:3.2.0]
```

## javac compile on ubuntu-latest / JDK latest-ea / mode current-defaults

Failure kind: javac compile
OS: ubuntu-latest
JDK: latest-ea
Native-image mode: current-defaults
Reproducer: `env GVM_TCK_LV=3.3.0 GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean compileTestJava -Pcoordinates=org.apache.maven.plugins:maven-source-plugin:3.2.0`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/26389440080
Last 300 lines of the log (excerpt truncated to fit the GitHub issue size limit):
```console
3.0 stage javac compile
========================================================================================
Starting stage 'javac compile' for 3.3.0 at 2026-05-25T07:53:21Z
Command: env GVM_TCK_LV=3.3.0 GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean compileTestJava -Pcoordinates=org.apache.maven.plugins:maven-source-plugin:3.2.0
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 1s
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :compileTestJava FAILED
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :tck-test-support:compileJava UP-TO-DATE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE

> Task :compileTestJava FAILED
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:9: error: package org.apache.maven.plugin.logging does not exist
import org.apache.maven.plugin.logging.Log;
                                      ^
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:10: error: cannot find symbol
import org.apache.maven.plugins.source.HelpMojo;
                                      ^
  symbol:   class HelpMojo
  location: package org.apache.maven.plugins.source
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:31: error: cannot find symbol
    private static final class CapturingLog implements Log {
                                                       ^
  symbol:   class Log
  location: class HelpMojoTest
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:18: error: cannot find symbol
        HelpMojo mojo = new HelpMojo();
        ^
  symbol:   class HelpMojo
  location: class HelpMojoTest
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:18: error: cannot find symbol
        HelpMojo mojo = new HelpMojo();
                            ^
  symbol:   class HelpMojo
  location: class HelpMojoTest
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:45: error: method does not override or implement a method from a supertype
        @Override
        ^
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:50: error: method does not override or implement a method from a supertype
        @Override
        ^
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:55: error: method does not override or implement a method from a supertype
        @Override
        ^
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:61: error: method does not override or implement a method from a supertype
        @Override
        ^
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:66: error: method does not override or implement a method from a supertype
        @Override
        ^
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:71: error: method does not override or implement a method from a supertype
        @Override
        ^
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:76: error: method does not override or implement a method from a supertype
        @Override
        ^
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:82: error: method does not override or implement a method from a supertype
        @Override
        ^
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:87: error: method does not override or implement a method from a supertype
        @Override
        ^
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:92: error: method does not override or implement a method from a supertype
        @Override
        ^
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:97: error: method does not override or implement a method from a supertype
        @Override
        ^
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:103: error: method does not override or implement a method from a supertype
        @Override
        ^
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:108: error: method does not override or implement a method from a supertype
        @Override
        ^
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:113: error: method does not override or implement a method from a supertype
        @Override
        ^
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:118: error: method does not override or implement a method from a supertype
        @Override
        ^
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:124: error: method does not override or implement a method from a supertype
        @Override
        ^
21 errors

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed; see the compiler output below.
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:10: error: cannot find symbol
  import org.apache.maven.plugins.source.HelpMojo;
                                        ^
    symbol:   class HelpMojo
    location: package org.apache.maven.plugins.source
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:31: error: cannot find symbol
      private static final class CapturingLog implements Log {
                                                         ^
    symbol:   class Log
    location: class HelpMojoTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:18: error: cannot find symbol
          HelpMojo mojo = new HelpMojo();
          ^
    symbol:   class HelpMojo
    location: class HelpMojoTest

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/build/reports/problems/problems-report.html
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:18: error: cannot find symbol
          HelpMojo mojo = new HelpMojo();
                              ^
    symbol:   class HelpMojo
    location: class HelpMojoTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:45: error: method does not override or implement a method from a supertype
          @Override
          ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:50: error: method does not override or implement a method from a supertype
          @Override
          ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:55: error: method does not override or implement a method from a supertype
          @Override
          ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:61: error: method does not override or implement a method from a supertype
          @Override
          ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:66: error: method does not override or implement a method from a supertype
          @Override
          ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:71: error: method does not override or implement a method from a supertype
          @Override
          ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:76: error: method does not override or implement a method from a supertype
          @Override
          ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:82: error: method does not override or implement a method from a supertype
          @Override
          ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:87: error: method does not override or implement a method from a supertype
          @Override
          ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:92: error: method does not override or implement a method from a supertype
          @Override
          ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:97: error: method does not override or implement a method from a supertype
          @Override
          ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:103: error: method does not override or implement a method from a supertype
          @Override
          ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:108: error: method does not override or implement a method from a supertype
          @Override
          ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:113: error: method does not override or implement a method from a supertype
          @Override
          ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:118: error: method does not override or implement a method from a supertype
          @Override
          ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:124: error: method does not override or implement a method from a supertype
          @Override
          ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:9: error: package org.apache.maven.plugin.logging does not exist
  import org.apache.maven.plugin.logging.Log;
                                        ^
  21 errors

* Try:
> Check your code and dependencies to fix the compilation error(s)
> Run with --scan to generate a Build Scan (Powered by Develocity).

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
10 actionable tasks: 1 executed, 9 up-to-date

BUILD FAILED in 1s

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 3s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'javac compile' for 3.3.0 at 2026-05-25T07:53:24Z with exit code 1 after 3s
❌ ERROR: 'javac compile' for 3.3.0 failed with exit code 1.
FAILED[javac compile][3.3.0][env GVM_TCK_LV=3.3.0 GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean compileTestJava -Pcoordinates=org.apache.maven.plugins:maven-source-plugin:3.2.0]
```

## javac compile on ubuntu-latest / JDK latest-ea / mode future-defaults-all

Failure kind: javac compile
OS: ubuntu-latest
JDK: latest-ea
Native-image mode: future-defaults-all
Reproducer: `env GVM_TCK_LV=3.3.0 GVM_TCK_NATIVE_IMAGE_MODE=future-defaults-all ./gradlew clean compileTestJava -Pcoordinates=org.apache.maven.plugins:maven-source-plugin:3.2.0`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/26389440080
Last 300 lines of the log (excerpt truncated to fit the GitHub issue size limit):
```console
ce-plugin:3.2.0:3.3.0 stage javac compile
========================================================================================
Starting stage 'javac compile' for 3.3.0 at 2026-05-25T07:53:04Z
Command: env GVM_TCK_LV=3.3.0 GVM_TCK_NATIVE_IMAGE_MODE=future-defaults-all ./gradlew clean compileTestJava -Pcoordinates=org.apache.maven.plugins:maven-source-plugin:3.2.0
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 844ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :compileTestJava FAILED
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :tck-test-support:compileJava UP-TO-DATE
> Task :classes UP-TO-DATE

> Task :compileTestJava FAILED
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:9: error: package org.apache.maven.plugin.logging does not exist
import org.apache.maven.plugin.logging.Log;
                                      ^
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:10: error: cannot find symbol
import org.apache.maven.plugins.source.HelpMojo;
                                      ^
  symbol:   class HelpMojo
  location: package org.apache.maven.plugins.source
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:31: error: cannot find symbol
    private static final class CapturingLog implements Log {
                                                       ^
  symbol:   class Log
  location: class HelpMojoTest
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:18: error: cannot find symbol
        HelpMojo mojo = new HelpMojo();
        ^
  symbol:   class HelpMojo
  location: class HelpMojoTest
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:18: error: cannot find symbol
        HelpMojo mojo = new HelpMojo();
                            ^
  symbol:   class HelpMojo
  location: class HelpMojoTest
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:45: error: method does not override or implement a method from a supertype
        @Override
        ^
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:50: error: method does not override or implement a method from a supertype
        @Override
        ^
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:55: error: method does not override or implement a method from a supertype
        @Override
        ^
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:61: error: method does not override or implement a method from a supertype
        @Override
        ^
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:66: error: method does not override or implement a method from a supertype
        @Override
        ^
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:71: error: method does not override or implement a method from a supertype
        @Override
        ^
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:76: error: method does not override or implement a method from a supertype
        @Override
        ^
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:82: error: method does not override or implement a method from a supertype
        @Override
        ^
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:87: error: method does not override or implement a method from a supertype
        @Override
        ^
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:92: error: method does not override or implement a method from a supertype
        @Override
        ^
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:97: error: method does not override or implement a method from a supertype
        @Override
        ^
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:103: error: method does not override or implement a method from a supertype
        @Override
        ^
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:108: error: method does not override or implement a method from a supertype
        @Override
        ^
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:113: error: method does not override or implement a method from a supertype
        @Override
        ^
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:118: error: method does not override or implement a method from a supertype
        @Override
        ^
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:124: error: method does not override or implement a method from a supertype
        @Override
        ^
21 errors

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed; see the compiler output below.
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:10: error: cannot find symbol
  import org.apache.maven.plugins.source.HelpMojo;
                                        ^
    symbol:   class HelpMojo
    location: package org.apache.maven.plugins.source
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:31: error: cannot find symbol
      private static final class CapturingLog implements Log {
                                                         ^
    symbol:   class Log
    location: class HelpMojoTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:18: error: cannot find symbol
          HelpMojo mojo = new HelpMojo();
          ^
    symbol:   class HelpMojo
    location: class HelpMojoTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:18: error: cannot find symbol
          HelpMojo mojo = new HelpMojo();
                              ^
    symbol:   class HelpMojo
    location: class HelpMojoTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:45: error: method does not override or implement a method from a supertype
          @Override
          ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:50: error: method does not override or implement a method from a supertype
          @Override
          ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:55: error: method does not override or implement a method from a supertype
          @Override
          ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:61: error: method does not override or implement a method from a supertype
          @Override
          ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:66: error: method does not override or implement a method from a supertype
          @Override
          ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:71: error: method does not override or implement a method from a supertype
          @Override
          ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:76: error: method does not override or implement a method from a supertype
          @Override
          ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:82: error: method does not override or implement a method from a supertype
          @Override
          ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:87: error: method does not override or implement a method from a supertype
          @Override
          ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:92: error: method does not override or implement a method from a supertype
          @Override
          ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:97: error: method does not override or implement a method from a supertype
          @Override
          ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:103: error: method does not override or implement a method from a supertype
          @Override
          ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:108: error: method does not override or implement a method from a supertype
          @Override
          ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:113: error: method does not override or implement a method from a supertype
          @Override
          ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:118: error: method does not override or implement a method from a supertype
          @Override
          ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:124: error: method does not override or implement a method from a supertype
          @Override
          ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-source-plugin/3.2.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java:9: error: package org.apache.maven.plugin.logging does not exist
  import org.apache.maven.plugin.logging.Log;
                                        ^
  21 errors

* Try:
> Check your code and dependencies to fix the compilation error(s)
> Run with --scan to generate a Build Scan (Powered by Develocity).

BUILD FAILED in 1s

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
10 actionable tasks: 1 executed, 9 up-to-date

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 3s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'javac compile' for 3.3.0 at 2026-05-25T07:53:07Z with exit code 1 after 3s
❌ ERROR: 'javac compile' for 3.3.0 failed with exit code 1.
FAILED[javac compile][3.3.0][env GVM_TCK_LV=3.3.0 GVM_TCK_NATIVE_IMAGE_MODE=future-defaults-all ./gradlew clean compileTestJava -Pcoordinates=org.apache.maven.plugins:maven-source-plugin:3.2.0]
```

--- a/metadata/org.apache.maven.plugins/maven-source-plugin/3.3.0/reachability-metadata.json
+++ b/metadata/org.apache.maven.plugins/maven-source-plugin/3.3.0/reachability-metadata.json
@@ -1,0 +1,10 @@
+{
+  "resources": [
+    {
+      "condition": {
+        "typeReached": "org.apache.maven.plugins.maven_source_plugin.HelpMojo"
+      },
+      "glob": "META-INF/maven/org.apache.maven.plugins/maven-source-plugin/plugin-help.xml"
+    }
+  ]
+}

--- a/metadata/org.apache.maven.plugins/maven-source-plugin/index.json
+++ b/metadata/org.apache.maven.plugins/maven-source-plugin/index.json
@@ -11,7 +11,8 @@
       "3.3.0"
     ],
     "allowed-packages" : [
-      "org.apache.maven.plugins.source"
+      "org.apache.maven.plugins.source",
+      "org.apache.maven.plugins.maven_source_plugin"
     ]
   },
   {

--- a/metadata/org.apache.maven.plugins/maven-source-plugin/index.json
+++ b/metadata/org.apache.maven.plugins/maven-source-plugin/index.json
@@ -1,6 +1,15 @@
 [
   {
     "latest" : true,
+    "metadata-version" : "3.3.0",
+    "tested-versions" : [
+      "3.3.0"
+    ],
+    "allowed-packages" : [
+      "org.apache.maven.plugins.source"
+    ]
+  },
+  {
     "metadata-version" : "3.2.0",
     "source-code-url" : "https://repo.maven.apache.org/maven2/org/apache/maven/plugins/maven-source-plugin/$version$/maven-source-plugin-$version$-sources.jar",
     "repository-url" : "https://github.com/apache/maven-source-plugin",

--- a/metadata/org.apache.maven.plugins/maven-source-plugin/index.json
+++ b/metadata/org.apache.maven.plugins/maven-source-plugin/index.json
@@ -2,6 +2,11 @@
   {
     "latest" : true,
     "metadata-version" : "3.3.0",
+    "source-code-url" : "https://repo.maven.apache.org/maven2/org/apache/maven/plugins/maven-source-plugin/$version$/maven-source-plugin-$version$-sources.jar",
+    "repository-url" : "https://github.com/apache/maven-source-plugin",
+    "test-code-url" : "https://repo.maven.apache.org/maven2/org/apache/maven/plugins/maven-source-plugin/$version$/maven-source-plugin-$version$-source-release.zip",
+    "documentation-url" : "https://repo.maven.apache.org/maven2/org/apache/maven/plugins/maven-source-plugin/$version$/maven-source-plugin-$version$-javadoc.jar",
+    "description" : "Apache Maven Source Plugin creates source and test-source archive artifacts for Maven projects. It provides goals to package project sources into JARs and attach them to the build for installation or deployment.",
     "tested-versions" : [
       "3.3.0"
     ],

--- a/stats/org.apache.maven.plugins/maven-source-plugin/3.3.0/execution-metrics.json
+++ b/stats/org.apache.maven.plugins/maven-source-plugin/3.3.0/execution-metrics.json
@@ -1,0 +1,99 @@
+{
+  "fix_javac_fail:2026-06-08": {
+    "timestamp": "2026-06-08T00:25:16.398759Z",
+    "library": "org.apache.maven.plugins:maven-source-plugin:3.3.0",
+    "starting_commit": "0784eeb9497f8c0cc548d2d83cbcb6173647a906",
+    "ending_commit": "11ecb8d8d7202d02256af88972472ca515b4c4a3",
+    "previous_library": "org.apache.maven.plugins:maven-source-plugin:3.2.0",
+    "strategy_name": "javac_iterative_with_coverage_sources_pi_gpt-5.5",
+    "agent": "pi",
+    "model": "oca/gpt-5.5",
+    "status": "success",
+    "stats": {
+      "version": "3.3.0",
+      "dynamicAccess": {
+        "breakdown": {
+          "resources": {
+            "coverageRatio": 1.0,
+            "coveredCalls": 1,
+            "totalCalls": 1
+          }
+        },
+        "coverageRatio": 1.0,
+        "coveredCalls": 1,
+        "totalCalls": 1
+      },
+      "libraryCoverage": {
+        "instruction": {
+          "covered": 538,
+          "missed": 1024,
+          "ratio": 0.34443,
+          "total": 1562
+        },
+        "line": {
+          "covered": 103,
+          "missed": 204,
+          "ratio": 0.335505,
+          "total": 307
+        },
+        "method": {
+          "covered": 14,
+          "missed": 32,
+          "ratio": 0.304348,
+          "total": 46
+        }
+      }
+    },
+    "previous_library_stats": {
+      "version": "3.2.0",
+      "dynamicAccess": {
+        "breakdown": {
+          "resources": {
+            "coverageRatio": 1.0,
+            "coveredCalls": 1,
+            "totalCalls": 1
+          }
+        },
+        "coverageRatio": 1.0,
+        "coveredCalls": 1,
+        "totalCalls": 1
+      },
+      "libraryCoverage": {
+        "instruction": {
+          "covered": 559,
+          "missed": 975,
+          "ratio": 0.364407,
+          "total": 1534
+        },
+        "line": {
+          "covered": 107,
+          "missed": 189,
+          "ratio": 0.361486,
+          "total": 296
+        },
+        "method": {
+          "covered": 14,
+          "missed": 31,
+          "ratio": 0.311111,
+          "total": 45
+        }
+      }
+    },
+    "metrics": {
+      "input_tokens_used": 45034,
+      "cached_input_tokens_used": 258560,
+      "output_tokens_used": 1712,
+      "iterations": 1,
+      "cost_usd": 0.1807,
+      "tested_library_loc": 103,
+      "metadata_entries": 1,
+      "previous_library_metadata_entries": 1,
+      "code_coverage_percent": 33.55,
+      "previous_library_coverage_percent": 36.15
+    },
+    "artifacts": {
+      "test_file": "tests/src/org.apache.maven.plugins/maven-source-plugin/3.3.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java",
+      "metadata_file": "metadata/org.apache.maven.plugins/maven-source-plugin/3.3.0/reachability-metadata.json"
+    }
+  }
+}

--- a/stats/org.apache.maven.plugins/maven-source-plugin/3.3.0/stats.json
+++ b/stats/org.apache.maven.plugins/maven-source-plugin/3.3.0/stats.json
@@ -1,0 +1,37 @@
+{
+  "versions" : [ {
+    "version" : "3.3.0",
+    "dynamicAccess" : {
+      "breakdown" : {
+        "resources" : {
+          "coverageRatio" : 1.0,
+          "coveredCalls" : 1,
+          "totalCalls" : 1
+        }
+      },
+      "coverageRatio" : 1.0,
+      "coveredCalls" : 1,
+      "totalCalls" : 1
+    },
+    "libraryCoverage" : {
+      "instruction" : {
+        "covered" : 538,
+        "missed" : 1024,
+        "ratio" : 0.34443,
+        "total" : 1562
+      },
+      "line" : {
+        "covered" : 103,
+        "missed" : 204,
+        "ratio" : 0.335505,
+        "total" : 307
+      },
+      "method" : {
+        "covered" : 14,
+        "missed" : 32,
+        "ratio" : 0.304348,
+        "total" : 46
+      }
+    }
+  } ]
+}

--- a/tests/src/org.apache.maven.plugins/maven-source-plugin/3.3.0/.gitignore
+++ b/tests/src/org.apache.maven.plugins/maven-source-plugin/3.3.0/.gitignore
@@ -1,0 +1,4 @@
+gradlew.bat
+gradlew
+gradle/
+build/

--- a/tests/src/org.apache.maven.plugins/maven-source-plugin/3.3.0/build.gradle
+++ b/tests/src/org.apache.maven.plugins/maven-source-plugin/3.3.0/build.gradle
@@ -12,6 +12,7 @@ String libraryVersion = tck.testedLibraryVersion.get()
 
 dependencies {
     testImplementation "org.apache.maven.plugins:maven-source-plugin:$libraryVersion"
+    testImplementation 'org.apache.maven:maven-plugin-api:3.2.5'
     testImplementation 'org.assertj:assertj-core:3.22.0'
 }
 

--- a/tests/src/org.apache.maven.plugins/maven-source-plugin/3.3.0/build.gradle
+++ b/tests/src/org.apache.maven.plugins/maven-source-plugin/3.3.0/build.gradle
@@ -1,0 +1,27 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+plugins {
+    id "org.graalvm.internal.tck"
+}
+
+String libraryVersion = tck.testedLibraryVersion.get()
+
+dependencies {
+    testImplementation "org.apache.maven.plugins:maven-source-plugin:$libraryVersion"
+    testImplementation 'org.assertj:assertj-core:3.22.0'
+}
+
+graalvmNative {
+    agent {
+        defaultMode = "conditional"
+        modes {
+            conditional {
+                userCodeFilterPath = "user-code-filter.json"
+            }
+        }
+    }
+}

--- a/tests/src/org.apache.maven.plugins/maven-source-plugin/3.3.0/gradle.properties
+++ b/tests/src/org.apache.maven.plugins/maven-source-plugin/3.3.0/gradle.properties
@@ -1,0 +1,6 @@
+# Optional explicit tested library coordinates (format: group:artifact:version).
+# If omitted, fallback resolution in org.graalvm.internal.tck.gradle still uses
+# environment variable GVM_TCK_LC and then this property.
+library.coordinates = org.apache.maven.plugins:maven-source-plugin:3.3.0
+library.version = 3.3.0
+metadata.dir = org.apache.maven.plugins/maven-source-plugin/3.3.0/

--- a/tests/src/org.apache.maven.plugins/maven-source-plugin/3.3.0/settings.gradle
+++ b/tests/src/org.apache.maven.plugins/maven-source-plugin/3.3.0/settings.gradle
@@ -1,0 +1,20 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+pluginManagement {
+    def tckPath = Objects.requireNonNullElse(
+            System.getenv("GVM_TCK_TCKDIR"),
+            "../../../../tck-build-logic"
+    )
+    includeBuild(tckPath)
+    repositories { mavenLocal(); gradlePluginPortal() }
+}
+
+plugins {
+    id "org.graalvm.internal.tck-settings" version "1.0.0-SNAPSHOT"
+}
+
+rootProject.name = 'org.apache.maven.plugins.maven-source-plugin_tests'

--- a/tests/src/org.apache.maven.plugins/maven-source-plugin/3.3.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java
+++ b/tests/src/org.apache.maven.plugins/maven-source-plugin/3.3.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java
@@ -1,0 +1,137 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+package org_apache_maven_plugins.maven_source_plugin;
+
+import org.apache.maven.plugin.logging.Log;
+import org.apache.maven.plugins.source.HelpMojo;
+import org.junit.jupiter.api.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class HelpMojoTest {
+    @Test
+    public void executeReadsBundledPluginHelpResource() throws Exception {
+        HelpMojo mojo = new HelpMojo();
+        CapturingLog log = new CapturingLog();
+        mojo.setLog(log);
+
+        mojo.execute();
+
+        assertThat(log.debugMessages()).contains("plugin-help.xml");
+        assertThat(log.infoMessages())
+                .contains("Maven Source Plugin")
+                .contains("source:jar")
+                .contains("source:test-jar");
+    }
+
+    private static final class CapturingLog implements Log {
+        private final StringBuilder debugMessages = new StringBuilder();
+        private final StringBuilder infoMessages = new StringBuilder();
+        private final StringBuilder warnMessages = new StringBuilder();
+        private final StringBuilder errorMessages = new StringBuilder();
+
+        String debugMessages() {
+            return debugMessages.toString();
+        }
+
+        String infoMessages() {
+            return infoMessages.toString();
+        }
+
+        @Override
+        public boolean isDebugEnabled() {
+            return true;
+        }
+
+        @Override
+        public void debug(CharSequence content) {
+            append(debugMessages, content);
+        }
+
+        @Override
+        public void debug(CharSequence content, Throwable error) {
+            append(debugMessages, content);
+            append(debugMessages, error);
+        }
+
+        @Override
+        public void debug(Throwable error) {
+            append(debugMessages, error);
+        }
+
+        @Override
+        public boolean isInfoEnabled() {
+            return true;
+        }
+
+        @Override
+        public void info(CharSequence content) {
+            append(infoMessages, content);
+        }
+
+        @Override
+        public void info(CharSequence content, Throwable error) {
+            append(infoMessages, content);
+            append(infoMessages, error);
+        }
+
+        @Override
+        public void info(Throwable error) {
+            append(infoMessages, error);
+        }
+
+        @Override
+        public boolean isWarnEnabled() {
+            return true;
+        }
+
+        @Override
+        public void warn(CharSequence content) {
+            append(warnMessages, content);
+        }
+
+        @Override
+        public void warn(CharSequence content, Throwable error) {
+            append(warnMessages, content);
+            append(warnMessages, error);
+        }
+
+        @Override
+        public void warn(Throwable error) {
+            append(warnMessages, error);
+        }
+
+        @Override
+        public boolean isErrorEnabled() {
+            return true;
+        }
+
+        @Override
+        public void error(CharSequence content) {
+            append(errorMessages, content);
+        }
+
+        @Override
+        public void error(CharSequence content, Throwable error) {
+            append(errorMessages, content);
+            append(errorMessages, error);
+        }
+
+        @Override
+        public void error(Throwable error) {
+            append(errorMessages, error);
+        }
+
+        private static void append(StringBuilder messages, CharSequence content) {
+            messages.append(content).append('\n');
+        }
+
+        private static void append(StringBuilder messages, Throwable error) {
+            messages.append(error.getClass().getName()).append(": ").append(error.getMessage()).append('\n');
+        }
+    }
+}

--- a/tests/src/org.apache.maven.plugins/maven-source-plugin/3.3.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java
+++ b/tests/src/org.apache.maven.plugins/maven-source-plugin/3.3.0/src/test/java/org_apache_maven_plugins/maven_source_plugin/HelpMojoTest.java
@@ -7,7 +7,7 @@
 package org_apache_maven_plugins.maven_source_plugin;
 
 import org.apache.maven.plugin.logging.Log;
-import org.apache.maven.plugins.source.HelpMojo;
+import org.apache.maven.plugins.maven_source_plugin.HelpMojo;
 import org.junit.jupiter.api.Test;
 
 import static org.assertj.core.api.Assertions.assertThat;

--- a/tests/src/org.apache.maven.plugins/maven-source-plugin/3.3.0/user-code-filter.json
+++ b/tests/src/org.apache.maven.plugins/maven-source-plugin/3.3.0/user-code-filter.json
@@ -7,6 +7,9 @@
       "includeClasses" : "org.apache.maven.plugins.source.**"
     },
     {
+      "includeClasses" : "org.apache.maven.plugins.**"
+    },
+    {
       "includeClasses" : "org_apache_maven_plugins.maven_source_plugin.**"
     }
   ]

--- a/tests/src/org.apache.maven.plugins/maven-source-plugin/3.3.0/user-code-filter.json
+++ b/tests/src/org.apache.maven.plugins/maven-source-plugin/3.3.0/user-code-filter.json
@@ -1,0 +1,13 @@
+{
+  "rules" : [
+    {
+      "excludeClasses" : "**"
+    },
+    {
+      "includeClasses" : "org.apache.maven.plugins.source.**"
+    },
+    {
+      "includeClasses" : "org_apache_maven_plugins.maven_source_plugin.**"
+    }
+  ]
+}

